### PR TITLE
Added dev_note_npm in a similar fashion to the original notebook app

### PR DESCRIPTION
**Pull request by jdetle**
Addresses #438. 

This follows the style of the original notebook.

### CODE REVIEW

**Comment by jdetle** [comment]
cc @jasongrout, @afshin, @sccolbert, @blink1073

File: jupyterlab/__init__.py [hunk 1]
```diff
@@ -45,9 +61,13 @@ def _jupyter_server_extension_paths():
...
     }]
-    
+
 
 def load_jupyter_server_extension(nbapp):
+    base_dir = os.path.realpath(os.path.join(HERE, '..'))
+    dev_mode = os.path.exists(os.path.join(base_dir, '.git'))
+    if dev_mode:
+        nbapp.log.info(DEV_NOTE_NPM)
     nbapp.log.info('Pre-alpha version of JupyterLab extension loaded from %s'%HERE)
```
Comment (jasongrout) [inline jupyterlab/__init__.py]: Can you fix this to say 'JupyterLab alpha preview extension ...' while you're editing this code?

Comment (jdetle) [inline jupyterlab/__init__.py]: Done!

File: jupyterlab/__init__.py [hunk 2]
```diff
@@ -9,13 +9,29 @@
 from jinja2 import FileSystemLoader
 from notebook.utils import url_path_join as ujoin
 
+
+#-----------------------------------------------------------------------------
+# Module globals
+#-----------------------------------------------------------------------------
+
+DEV_NOTE_NPM = """It looks like you're running JupyterLab from source.
+If you're working on the TypeScript of the notebook, try running
```
Comment (jasongrout) [inline jupyterlab/__init__.py]: If you're working on the Typescript sources of JupyterLab ....

File: jupyterlab/__init__.py [hunk 3]
```diff
@@ -9,13 +9,29 @@
...
+
+    npm run watch
+
+in another terminal window to have the system incrementally
```
Comment (jasongrout) [inline jupyterlab/__init__.py]: from the JupyterLab repo directory in another terminal window...

**Comment by jasongrout** [comment]
Looks great, module two small wording suggestions...

File: jupyterlab/__init__.py (hunk 2 above)
Comment (jdetle) [inline jupyterlab/__init__.py]: Oh shoot, sorry for the simple mistake.

File: jupyterlab/__init__.py (hunk 3 above)
Comment (jasongrout) [inline jupyterlab/__init__.py]: or maybe "in the JupyterLab repo directory from another terminal window..."

**Comment by jasongrout** [comment]
Looks good. Thanks!

**Pull request merged**